src/test/streaming_test.cpp

```diff
@@ -0,0 +1,650 @@
+    ASSERT_TRUE(::google::protobuf::TextFormat::ParseFromString(pbTxt, &config));
+
+    MediapipeGraphExecutor executor{
+        this->name, this->version, config,
+        {{"in1", mediapipe_packet_type_enum::OVTENSOR}, {"in2", mediapipe_packet_type_enum::OVTENSOR}, {"in3", mediapipe_packet_type_enum::OVTENSOR}},
+        {{"out1", mediapipe_packet_type_enum::OVTENSOR}, {"out2", mediapipe_packet_type_enum::OVTENSOR}, {"out3", mediapipe_packet_type_enum::OVTENSOR}},
+        {"in1", "in2", "in3"}, {"out1", "out2", "out3"}, {}};
+
+    std::mutex mtx;
+    const int64_t timestamp = 64;
+
+    prepareRequest(this->firstRequest, {{"in1", 3.5f}}, timestamp);
+    EXPECT_CALL(this->stream, Read(_))
+        .WillOnce(ReceiveWithTimestamp({{"in2", 7.2f}}, timestamp))
+        .WillOnce(ReceiveWithTimestamp({{"in3", 102.4f}}, timestamp))
+        .WillOnce(DisconnectWhenNotified(mtx));
+
+    EXPECT_CALL(this->stream, Write(_, _))
+        .WillOnce(SendWithTimestamp({{"out1", 4.5f}}, timestamp))
+        .WillOnce(SendWithTimestamp({{"out2", 8.2f}}, timestamp))
+        .WillOnce(SendWithTimestampAndNotifyEnd({{"out3", 103.4f}}, timestamp, mtx));
+
+    ASSERT_EQ(executor.inferStream(this->firstRequest, this->stream), StatusCode::OK);
+}
+
+// Sending inputs together for synchronized graph
+TEST_F(StreamingTest, MultipleStreamsDeliveredViaSingleRequest) {
+    const std::string pbTxt{R"(
+input_stream: "in1"
+input_stream: "in2"
+input_stream: "in3"
+output_stream: "out1"
+output_stream: "out2"
+output_stream: "out3"
+node {
+  calculator: "StreamingMultiInputsOutputsTestCalculator"
+  input_stream: "in1"
+  input_stream: "in2"
+  input_stream: "in3"
+  output_stream: "out1"
+  output_stream: "out2"
+  output_stream: "out3"
+}
+    )"};
+    ::mediapipe::CalculatorGraphConfig config;
+    ASSERT_TRUE(::google::protobuf::TextFormat::ParseFromString(pbTxt, &config));
+
+    MediapipeGraphExecutor executor{
+        this->name, this->version, config,
+        {{"in1", mediapipe_packet_type_enum::OVTENSOR}, {"in2", mediapipe_packet_type_enum::OVTENSOR}, {"in3", mediapipe_packet_type_enum::OVTENSOR}},
+        {{"out1", mediapipe_packet_type_enum::OVTENSOR}, {"out2", mediapipe_packet_type_enum::OVTENSOR}, {"out3", mediapipe_packet_type_enum::OVTENSOR}},
+        {"in1", "in2", "in3"}, {"out1", "out2", "out3"}, {}};
+
+    std::mutex mtx;
+    const int64_t timestamp = 64;
+
+    prepareRequest(this->firstRequest, {{"in1", 3.5f}, {"in2", 7.2f}, {"in3", 102.4f}}, timestamp);
+    EXPECT_CALL(this->stream, Read(_))
+        .WillOnce(DisconnectWhenNotified(mtx));
+
+    EXPECT_CALL(this->stream, Write(_, _))
+        .WillOnce(SendWithTimestamp({{"out1", 4.5f}}, timestamp))
+        .WillOnce(SendWithTimestamp({{"out2", 8.2f}}, timestamp))
+        .WillOnce(SendWithTimestampAndNotifyEnd({{"out3", 103.4f}}, timestamp, mtx));
+
+    ASSERT_EQ(executor.inferStream(this->firstRequest, this->stream), StatusCode::OK);
```

Comment:
In next pull request (integration) - send error when model_name()/model_version() is different than firstRequest (but do not disconnect)